Tools/Algorithms/stencil.py

```diff
@@ -130,112 +133,127 @@ def compute_stencils(coeff_nodal, coeff_stagg, axis):
 
-def compute_all(dx, dy, dz, dt, nox, noy, noz, v_gal, nx=256, ny=256, nz=256):
+def compute_all(dx_boosted, dt, psatd_order, v_gal, nx=None):
     """
     Compute nodal and staggered stencils along all directions.
 
     Parameters
     ----------
-    dx : float
-        Cell size along x.
-    dy : float
-        Cell size along y.
-    dz : float
-        Cell size along z.
+    dx_boosted : np.ndarray (float)
+        Cell size along each direction.
     dt : float
         Time step.
-    nox : int
-        Spectral order along x.
-    noy : int
-        Spectral order along y.
-    noz : int
-        Spectral order along z.
+    psatd_order : np.ndarray (int)
+        Spectral order along each direction.
     v_gal : float
         Galilean velocity.
-    nx : int, optional (default = 256)
-        Number of mesh points along x.
-    ny : int, optional (default = 256)
-        Number of mesh points along y.
-    nz : int, optional (default = 256)
-        Number of mesh points along z.
+    nx : np.ndarray (int), optional (default = 256 in each direction)
+        Number of mesh points along each direction.
 
     Returns
     -------
-    (stencil_nodal, stencil_stagg) : tuple
-        Nodal and staggered stencils along all directions.
+    stencils : list
+        List of nodal and staggered stencils along all directions.
     """
-    # k vectors
-    kx_arr = 2*np.pi*np.fft.fftfreq(nx, dx)
-    ky_arr = 2*np.pi*np.fft.fftfreq(ny, dy)
-    kz_arr = 2*np.pi*np.fft.fftfreq(nz, dz)
-
-    # Centered modified k vectors
-    kx_arr_c = modified_k(kx_arr, dx, nox, False) if nox != 'inf' else kx_arr
-    ky_arr_c = modified_k(ky_arr, dy, noy, False) if noy != 'inf' else ky_arr
-    kz_arr_c = modified_k(kz_arr, dz, noz, False) if noz != 'inf' else kz_arr
-
-    # Staggered modified k vectors
-    kx_arr_s = modified_k(kx_arr, dx, nox, True) if nox != 'inf' else kx_arr
-    ky_arr_s = modified_k(ky_arr, dy, noy, True) if noy != 'inf' else ky_arr
-    kz_arr_s = modified_k(kz_arr, dz, noz, True) if noz != 'inf' else kz_arr
+    # Number of dimensions
+    dims = len(dx_boosted)
+
+    # Default value for nx
+    if nx is None:
+        nx = np.full(shape=dims, fill_value=256)
+
+    # k vectors and modified k vectors
+    k_arr   = []
+    k_arr_c = []
+    k_arr_s = []
+    for i in range(dims):
+        k_arr.append(2*np.pi*np.fft.fftfreq(nx[i], dx_boosted[i]))
+        if psatd_order[i] != 'inf':
+            k_arr_c.append(modified_k(k_arr[i], dx_boosted[i], psatd_order[i], False))
+            k_arr_s.append(modified_k(k_arr[i], dx_boosted[i], psatd_order[i], True))
+        else:
+            k_arr_c.append(k_arr[i])
+            k_arr_s.append(k_arr[i])
 
     # Mesh in k space
-    kx_c, ky_c, kz_c = np.meshgrid(kx_arr_c, ky_arr_c, kz_arr_c)
-    kx_s, ky_s, kz_s = np.meshgrid(kx_arr_s, ky_arr_s, kz_arr_s)
+    k_c = np.meshgrid(*k_arr_c)
+    k_s = np.meshgrid(*k_arr_s)
+    kk_c = 0.
+    kk_s = 0.
+    for k in k_c:
```

Comment:
How about
`kk_c = (k_c**2).sum()`

Reply:
I went for this solution, it is still better than my previous implementation:
```py
kk_c = np.sqrt(sum(k**2 for k in k_c))
```
The other solution didn't seem to work, I think because the operator `**` was mixing a list (`k_c`) and an int (the exponent).